File: Nov2020/16Nov/MaximizeTheEarning.java
/*
Problem Name: Maximize the Earning
Problem Link: https://www.hackerearth.com/practice/data-structures/arrays/1-d/practice-problems/algorithm/maximize-the-earning-137963bc-323025a6/
*/
import java.util.*;
import java.io.*;
class MaximizeTheEarning {
    public static void main(String args[] ) throws Exception {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        int s = Integer.parseInt(in.readLine());
        for(;s>0;s--){
            String[] str = in.readLine().split(" ");
            int n = Integer.parseInt(str[0]), r = Integer.parseInt(str[1]), earning = 0, max = 0;
            str = in.readLine().split(" ");
            for(int i=0;i<n;i++){
                int cur = Integer.parseInt(str[i]);
                if(max < cur){
                    earning += r;
                    max = cur;
                }
            }
            System.out.println(earning);
        }
    }
}

File: Nov2020/16Nov/MaximizeTheEarning.cpp
/*
Problem Name: Maximize the Earning
Problem Link: https://www.hackerearth.com/practice/data-structures/arrays/1-d/practice-problems/algorithm/maximize-the-earning-137963bc-323025a6/
*/

#include<bits/stdc++.h>
using namespace std;
int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    int S;
    cin>>S;
    while(S--){
        int N,R;
        cin>>N>>R;
        int H[N],b=1,max;
        cin>>H[0];

        max=H[0];
        for(int i=1;i<N;++i){
            cin>>H[i];
            if(max<H[i]){
                max=H[i];
                b++;
            }
        }
        cout<<b*R<<endl;
    }
    return 0;
}
